### ISSUE TITLE
Opening inventory GUI and hidden equipped weapons

### ISSUE THREAD

**Issue by Shad000w** [Beamdog/nwn-issues#532]
**To Reproduce**

 - Equip any melee weapon.
 - Hide it with SetHiddenWhenEquipped (needs to be manually using command or 0.3 after OnEquip event for this to work)
 - Now the character sheet says "Unarmed" and character will use monk attack animations.
 - Close inventory, if open. Open inventory.
 - Character will perform some animation with his right hand and the character sheet (needs to be re-opened) will now say "Whatever Weapon you used" instead of unarmed. And the animations will change back to classic melee weapon ones.

**Specifics**

 - OS: Any
 - Game build: Any
 - Distro platform: irrelevant
 - Content overrides: none
 - Regression: no

**If needed, describe the bug**

I am not sure if this behavior (using unarmed animations) is intended or not when the weapon is hidden, but in case it is, then it is kinda useless if opening inventory resets it everytime. There is also no inventory GUI open event to re-hide it to workaround this. What I want to do is to put gloves into right hand slot to free up arms slot for monks. With some additional tweaks in nwnx, this works as I want, but the animation issue is very annoying and really screws this feature.

Let me know if you consider this to be something to solve, if so I can upload a repro module.

**Comment by Shad000w** [comment]
> Definitely a bug, but I'm not sure if the actual behavior should be monk/unarmed animations or actual weapon animations just without the weapon.
> 
> There's a nwnx event that fires on inventory open, by the way, if you want a workaround.

Yes that works as workaround. Running script when opening inventory and unhide and re-hide the item workarounds my problem.

And to clarify. In my opinion the problem is that opening inventory for some reason changes the weapon animations. That seems to be a client-side thing as that happens even if I block the request on server side.

SetHiddenWhenEquipped actually works both way. If the item is hidden immediately after equip, with no delay, it will use original weapon animations but with nothing in hand. It results to unarmed animations only when that hiding is done after 0.3 seconds or later. Which will most likely be another bug as same function should not have 2 different outcomes, but I like this bug :) as it can be abused for what I want to achieve.

**Comment by mtijanic** [comment]
FYI you can also achieve what you want with ReplaceObjectAnimation() to change the weapon attack animation to the unarmed one.

**Comment by Shad000w** [comment]
> FYI you can also achieve what you want with ReplaceObjectAnimation() to change the weapon attack animation to the unarmed one.

Thanks. I am on a build that doesn't have this, but I will keep this in mind.

**Comment by Shad000w** [comment]
Update: same thing (the hand animation and unarmed animations being reset to one-hand with nothing in hand) when closing container.

We were able to reliably reproduce it with these steps:
- open placeable inventory
- open container in PC inventory (placeale inventory is closed automatically)
- use container unique power (or likely any other Cast Spell you give it to it)
- close container
